int64 col
int64 row
---
int64 neighbors